src/background/refreshMenus.ts

```diff
@@ -31,14 +31,42 @@ export async function refreshMenus() {
       contexts: ["selection"],
       title: `Send selection to ${chatName} ${chat.type}`,
       onclick: async (info, tab) => {
-        const data = await sendTabMessage(tab.id!, "getSelection", undefined)
+        const data = info.selectionText
         if (data == null) return
 
-        await sendMediaGroup(botToken, { chat_id: chatId, ...data }).catch((error) => {
+        await sendMessage(botToken, { chat_id: chatId, text: data }).catch((error) => {
           showError(`Sending selection to ${chatName} ${chat.type} failed: ${error.message}`)
         })
       },
     })
+
+    browser.menus.create({
+      id: `sendLink-${botToken}-${chatId}`,
+      contexts: ["link"],
+      title: `Send link to ${chatName} ${chat.type}`,
+      onclick: async (info, tab) => {
+        const data = info.linkUrl
+        if (data == null) return
+
+        await sendMessage(botToken, { chat_id: chatId, text: data }).catch((error) => {
+          showError(`Sending link to ${chatName} ${chat.type} failed: ${error.message}`)
+        })
+      },
+    })
+
+    browser.menus.create({
+      id: `sendPage-${botToken}-${chatId}`,
+      contexts: ["page"],
+      title: `Send page to ${chatName} ${chat.type}`,
+      onclick: async (info, tab) => {
+        const data = tab.url
+        if (data == null) return
+
+        await sendMessage(botToken, { chat_id: chatId, text: data }).catch((error) => {
+          showError(`Sending page to ${chatName} ${chat.type} failed: ${error.message}`)
+        })
+      },
+    })
```

Comment:
Instead of putting "Send page" in a context menu I'd made it a page action which displays in the location bar. The manifest already defines "Send to Telegram" page action and there's some code for handling it in `src/background/index.ts` but it's commented out.

Simply move the logic of `onclick` from there to `browser.pageAction.onClicked` handler in background index.

Reply:
If this were moved into a page action there'd need to be a way to specify which group to send the page to, should that be a new configuration option?

Reply:
You're right... The page action would need a html popup with the list of channels.